design/one-pager-generic-usage-type.md

```diff
@@ -0,0 +1,397 @@
+other systems - notably Terraform - avoid this by computing a dependency graph
+before attempting to enact the desired state. This allows the system to avoid
+trying operations it knows will fail due to missing dependencies until those
+dependencies are expected to be satisfied.
+
+A frequent negative side-effect of the eventually consistent approach that we
+see in Crossplane is resources becoming orphaned at delete time because their
+successful deletion depends on the existence of other resources which are often
+deleted before their dependents. It’s possible to author a set of interdependent
+resources that will eventually create successfully, but may not eventually
+delete successfully.
+
+Two variants of this problem are:
+
+- **Orphaned Managed Resources (MRs).** Consider a Helm Release MR that is
+deployed to an EKS Cluster MR. The Helm Release has a dependency on the EKS
+Cluster’s connection details - it uses them as Provider credentials. If the EKS
+Cluster is deleted before the Helm Release the Helm Release controller becomes
+unable to connect to its Provider in order to delete the Release. In practice
+the Release External Resource (ER) was implicitly deleted along with the GKE
+Cluster - there’s actually no work left for the Helm release controller to do -
+but it cannot know that. This results in the Helm Release MR being ‘orphaned’
+in a pending deletion that can never be satisfied.
+
+- **Orphaned ‘Side-Effect’ Resources.** Consider the same example as above, and
+assume the Helm Release includes a Kubernetes Service resource of type:
+LoadBalancer. A side-effect of the creation of this Service will be to create an
+ELB in the EKS cluster’s VPC. This ELB is not managed by or even known to
+Crossplane, and is not deleted along with the EKS cluster. This means that the
+Helm Release (and therefore its Service) must be deleted before the EKS cluster
+in order to ensure the EKS cluster’s controllers trigger deletion of the ELB.
+If it is not, the ELB will be orphaned, costing money and blocking the deletion
+of the VPC it was created in.
+
+These problems are exacerbated by the fact that Kubernetes uses "background
+cascading deletion" by default. Put otherwise, the successful deletion of a
+parent resource is not dependent on the successful deletion of its children.
+Instead, the parent resource will be deleted successfully and the children
+(i.e. resources that declare the parent as an owner reference) will later be
+deleted by the Kubernetes garbage collector. This makes it difficult to diagnose
+or even notice resources that become orphaned when their parent XR is deleted.
+
+## Goals
+
+- Support "cross-provider" dependencies, for example an MR from provider-helm
+depending on a ProviderConfig from provider-aws.
+- Don’t require Crossplane users to manage fine-grained RBAC access.
+- Don’t grant Crossplane providers broad RBAC access (e.g. to all other providers).
```

Comment:
Adding brief note below explaining how the proposal achieves this would probably be useful. i.e. The RBAC manager already grants Crossplane RBAC access to all types in order for it to do Composition.